R/g.impute.R

```diff
@@ -135,7 +207,7 @@ g.impute = function(M, I, params_cleaning = c(), desiredtz = "",
     r2tempe = rep(r2, each = (ws2/ws3))
     r1tempe = rep(r1, each = (ws2/ws3))
     atest[which(r2tempe == 1 | r1tempe == 1)] = 0
-    if (params_cleaning[["strategy"]] == 3) {
+    if (params_cleaning[["data_masking_strategy"]] == 3) {
       # Select the most active 24-h blocks by a rolling window of windowsizes[3]
       NDAYS = length(atest) / n_ws3_perday
       # rolling window in ws3
```

Comment:
In lines 150:154, `r4` is cut out so that it only contains the data selected by the `study_dates_file` (only when `study_dates_file` is provided). The full `r4` vector is then stored in `r4_bu` to be recovered later on in lines 328:332. This avoids substantial changes in the code for every strategy. Everything occurs on the top of the strategies, so that previous functionality is not affected if not working with `study_dates_file`, and it is also easier to traceback potential problems when using this new functionality.